commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java

```diff
@@ -0,0 +1,171 @@
+        if (!Files.isDirectory(sourceDir)) {
+            return false;
+        }
+        String prefix = computeConnectClientPrefix(config);
+        boolean defaultPrefix = "connect".equals(prefix);
+        Path customClient = sourceDir.resolve("connect-client.ts");
+        if (Files.exists(customClient)) {
+            try {
+                String content = Files.readString(customClient);
+                if (!content.contains("const client = new ConnectClient({prefix: '" + prefix + "'});")) {
+                    LOGGER.debug(
+                            "Custom connect-client.ts detected ({}), but prefix does not match configuration {}.",
+                            customClient,
+                            prefix);
+                    return true;
+                }
+            } catch (IOException e) {
+                LOGGER.debug("Custom connect-client.ts detected ({}), but cannot read content.", customClient);
+                return false;
+            }
+            LOGGER.debug("Custom connect-client.ts detected ({}) with expected prefix {}.", customClient, prefix);
+            return false;
+        } else if (!defaultPrefix) {
+            LOGGER.debug("Custom prefix {} detected, connect-client.ts to be created in {}.", prefix, customClient);
+        }
+        return !defaultPrefix;
+    }
+
+    @Override
+    public boolean trigger(CodeGenContext context) {
+        String prefix = computeConnectClientPrefix(context.config());
+        boolean defaultPrefix = "connect".equals(prefix);
+        Path customClient = context.inputDir().resolve("connect-client.ts");
+        if (Files.exists(customClient)) {
+            String content = null;
+            try {
+                content = Files.readString(customClient);
+            } catch (IOException e) {
+                LOGGER.debug(
+                        "Cannot read content of custom connect-client.ts ({}). File will not be overwritten.",
+                        customClient);
+            }
+            if (content != null) {
+                content = content.replaceFirst(
+                        "(const client = new ConnectClient\\(\\{prefix:\\s*')[^']+('}\\);)", "$1" + prefix + "$2");
+                try {
+                    Files.writeString(customClient, content);
+                    LOGGER.debug(
+                            "Prefix in custom connect-client.ts ({}) replaced with new value {}", customClient, prefix);
+                    return true;
+                } catch (IOException e) {
+                    LOGGER.debug("Cannot write content of custom connect-client.ts ({}).", customClient);
+                }
+            }
+        } else if (!defaultPrefix) {
+            return writeConnectClient(prefix, customClient);
+        }
+        return false;
+    }
+
+    private static String computeConnectClientPrefix(Config config) {
+        String prefix = config.getValue("vaadin.endpoint.prefix", String.class);
```

Comment:
Make the config key static and reuse it where possible or use an config interface for it and inject it.

Reply:
I'm not sure inject is available during code generation (it runs also as a maven goal before build step processing). But static constant makes sense.